keytree/src/lib.rs

```diff
@@ -48,6 +52,35 @@ impl Xprv {
     }
 }
 
+impl Xpub {
+    /// Returns a intermediate child pubkey.
+    pub fn derive_intermediate_key(&self, customize: impl FnOnce(&mut Transcript)) -> Xpub {
+        let mut t = Transcript::new(b"Keytree.derivation");
+        t.commit_bytes(b"pt", self.precompressed_pubkey.as_bytes());
+        t.commit_bytes(b"dk", &self.dk);
+
+        // the user can pass customize in so that they can commit arbitrary
```

Comment:
i'd change this comment to reflect better that the "derivation path" is provided only through this closure. Meaning, it's not something that the user "can use" as an optionality, but "must use" in order to separate siblings from each other. I don't know what's the best way to put it yet.

Reply:
I added something to the function-level comment. Is that better?

Reply:
Yes!